improver_tests/cli/test_init.py

```diff
@@ -165,113 +166,94 @@ def test_with_output(self, m):
-        """Tests that it returns a function which itself returns 2 cubes"""
-        result = create_constrained_inputcubelist_converter(
-            "wind_speed", "wind_from_direction"
+    def test_basic(self):
+        """Tests a basic creation of create_constrained_inputcubelist_converter"""
+        func = create_constrained_inputcubelist_converter(
+            lambda cube: cube.name() in ["wind_speed"],
+            lambda cube: cube.name() in ["wind_from_direction"],
         )
-        result("foo")
-        m.assert_any_call(load_cube, "foo", constraints="wind_speed")
-        m.assert_any_call(load_cube, "foo", constraints="wind_from_direction")
-        self.assertEqual(m.call_count, 2)
-
-    @patch("improver.cli.maybe_coerce_with", return_value="return")
-    def test_list(self, m):
-        """Tests that a list returns a function which itself returns 2 cubes"""
-        result = create_constrained_inputcubelist_converter(
-            ["wind_speed"], ["wind_from_direction"]
+        result = func(self.wind_cubes)
+        self.assertIn(self.wind_speed_cube, result)
+        self.assertIn(self.wind_dir_cube, result)
+        self.assertEqual(2, len(result))
+
+    @patch("improver.cli.maybe_coerce_with", return_value=setup_for_mock())
+    def test_basic_given_str(self, mocked_maybe_coerce):
+        """Tests that a str is given to maybe_coerce_with which would return a CubeList."""
+        func = create_constrained_inputcubelist_converter(
+            lambda cube: cube.name() in ["wind_speed"],
+            lambda cube: cube.name() in ["wind_from_direction"],
         )
-        result("foo")
-        m.assert_any_call(load_cube, "foo", constraints="wind_speed")
-        m.assert_any_call(load_cube, "foo", constraints="wind_from_direction")
-        self.assertEqual(m.call_count, 2)
-
-    @patch("improver.cli.maybe_coerce_with", new=replace_load_with_extract)
-    def test_list_one_valid(self):
-        """Tests that a list returns a function which itself returns 1 cube
-        that matches the constraints provided."""
-        obj = create_constrained_inputcubelist_converter(["wind_speed", "nonsense"])
-        result = obj(self.wind_speed_cube)
-        self.assertEqual(result, [self.wind_speed_cube])
-
-    @patch("improver.cli.maybe_coerce_with", new=replace_load_with_extract)
+        result = func(self.wind_cubes)
+        self.assertIn(self.wind_speed_cube, result)
+        self.assertIn(self.wind_dir_cube, result)
+        self.assertEqual(2, len(result))
+        mocked_maybe_coerce.assert_called_once()
+
     def test_list_two_valid(self):
-        """Tests that providing two valid constraints raises a ValueError."""
-        obj = create_constrained_inputcubelist_converter(
-            ["wind_speed", "wind_from_direction"]
+        """Tests that one cube is loaded from each list."""
+        func = create_constrained_inputcubelist_converter(
+            lambda cube: cube.name() in ["speed", "wind_speed"],
```

Comment:
It might be more obvious to use something like "dummy" rather than "speed", so that it's more obvious that I wouldn't expect anything to be returned when using this name.